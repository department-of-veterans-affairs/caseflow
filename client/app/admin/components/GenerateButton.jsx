/* eslint-disable func-style */
import React, { useEffect, useState } from 'react';
import Button from '../../components/Button';
import Modal from '../../components/Modal';
import Alert from '../../components/Alert';
import LoadingContainer from '../../components/LoadingContainer';
import { LOGO_COLORS } from 'app/constants/AppConstants';
import PropTypes from 'prop-types';
import { CSVLink } from 'react-csv';

const GenerateButton = (props) => {

  // state properties
  const [showConfirmModal, setShowConfirmModal] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [showEmptyResultsBanner, setShowEmptyResultsBanner] = useState(false);
  const [showErrorBanner, setShowErrorBanner] = useState(false);
  const [extractedResults, setExtractedResults] = useState('');

  useEffect(() => {
    setIsLoading(props.isLoading);
  }, [props.isLoading]);

  useEffect(() => {
    setExtractedResults(props.extractedResults);

    if (props.extractedResults?.length > 0) {
      setShowEmptyResultsBanner(false);
      setShowConfirmModal(true);
    }
  }, [props.extractedResults]);

  useEffect(() => {
    if (props.emptyResultsMessage?.length > 0) {
      setShowEmptyResultsBanner(true);
      setShowConfirmModal(false);
    }
  }, [props.emptyResultsMessage]);

  useEffect(() => {
    if (props.manualExtractionSuccess === false) {
      setShowErrorBanner(true);
    } else {
      setShowErrorBanner(false);
    }
  }, [props.manualExtractionSuccess]);

  const onClickGenerate = () => {
    props.sendExtractRequest();
  };

  return (
    <div style={{ height: '75vh' }}>
      {
        showEmptyResultsBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="No Veterans were found" type="success" />
        </div>
      }
      {
        showErrorBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="Veteran Extract Failed" type="error" />
        </div>
      }
      {
        !isLoading &&
        <Button
          id="generate-extract"
          onClick={() => onClickGenerate()}
        >
          Generate Veteran Extract
        </Button>
      }
      {
        showConfirmModal &&

        <Modal title="This file contains PII"
          confirmButton={<CSVLink data={extractedResults} filename="veteran_extract.csv" onClick={() => setShowConfirmModal(false)}>Confirm</CSVLink>}
          closeHandler={() => {
            setShowConfirmModal(false);
          }}
        >
          Are you sure you want to download it?
        </Modal>
      }
      {
        isLoading &&

        <LoadingContainer color={LOGO_COLORS.QUEUE.ACCENT}>
          <div className="loading-div">
            Action is Running...
          </div>
        </LoadingContainer>
      }
    </div>
  );
};

GenerateButton.propTypes = {
  extractedResults: PropTypes.string,
  manualExtractionSuccess: PropTypes.bool,
  isLoading: PropTypes.bool,
  sendExtractRequest: PropTypes.func,
  emptyResultsMessage: PropTypes.string,
};

export default GenerateButton;
