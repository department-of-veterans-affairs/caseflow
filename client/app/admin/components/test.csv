VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN