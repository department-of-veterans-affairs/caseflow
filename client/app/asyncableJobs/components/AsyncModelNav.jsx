import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import DropdownButton from '../../components/DropdownButton';

const DATE_TIME_FORMAT = 'ddd MMM DD HH:mm:ss YYYY';

export default class AsyncModelNav extends React.PureComponent {
  modelNameLinks = () => {
    const links = [];
    const modelNames = this.props.models.sort();
    const numLinks = modelNames.length;

    for (let modelName of modelNames) {
      const url = `/asyncable_jobs/${modelName}/jobs`;
      let modelLink;

      if (numLinks > 4) {
        modelLink = {
          title: modelName,
          target: url
        };
      } else {
        modelLink = <span key={modelName} className="cf-model-jobs-link"><a href={url}>{modelName}</a></span>;
      }

      links.push(modelLink);
    }

    return numLinks > 4 ? <DropdownButton lists={links} label="Filter by Job Type" /> : links;
  }

  render = () => {
    const currentPath = this.props.asyncableJobKlass ?
      `/asyncable_jobs/${this.props.asyncableJobKlass}/jobs` :
      '/jobs';

    return <div>
      <strong>Last updated:</strong> {moment(this.props.fetchedAt).format(DATE_TIME_FORMAT)}
      <div style={{ marginTop: '.5em' }}>
        <a style={{ marginRight: '.5em' }} href="/jobs" className="cf-link-btn">All jobs</a>
        {this.modelNameLinks()}
        <a style={{ float: 'right' }} href={`${currentPath}.csv`} className="cf-link-btn">Download as CSV</a>
      </div>
    </div>;
  }
}

AsyncModelNav.propTypes = {
  models: PropTypes.array,
  fetchedAt: PropTypes.string,
  asyncableJobKlass: PropTypes.string
};
