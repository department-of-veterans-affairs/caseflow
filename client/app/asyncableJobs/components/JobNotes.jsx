import React from 'react';
import Button from '../../components/Button';
import ApiUtil from '../../util/ApiUtil';
import PropTypes from 'prop-types';
import moment from 'moment';
import ReactMarkdown from 'react-markdown';

const DATE_TIME_FORMAT = 'ddd MMM DD HH:mm:ss YYYY';

import Checkbox from '../../components/Checkbox';
import TextareaField from '../../components/TextareaField';

class NewNoteForm extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      saveInProgress: false,
      checkboxSelected: true,
      newNote: ''
    };
  }

  onSave = () => {
    const job = this.props.job;
    const url = `/asyncable_jobs/${job.klass}/jobs/${job.id}/note`;
    const form = this;
    const notes = this.props.notes;

    const data = {
      note: this.state.newNote,
      send_to_intake_user: this.state.checkboxSelected
    };

    notes.setState({ reloaded: false });

    ApiUtil.post(url, { data }).
      then(
        (response) => {
          const newNote = JSON.parse(response.text);

          form.setState({
            saveInProgress: false,
            checkboxSelected: false,
            newNote: ''
          });

          // add note to current page.
          notes.props.notes.unshift(newNote);
          notes.setState({ reloaded: true });
        }
      );
  }

  onChange = (event) => {
    this.setState({ newNote: event });
  }

  onCheckboxChange = (event) => {
    this.setState({ checkboxSelected: event });
  }

  isSaveDisabled = () => {
    return this.state.saveInProgress || this.state.newNote.trim().length === 0;
  }

  isCheckboxChecked = () => {
    return this.state.checkboxSelected;
  }

  sendToUserCheckbox = () => {
    return <Checkbox
      label="Send as message to user"
      name="send_to_intake_user"
      value={this.isCheckboxChecked()}
      onChange={this.onCheckboxChange}
      disabled={this.state.saveInProgress}
    />;
  }

  render = () => {
    return <div className="comment-size-container">
      <TextareaField
        hideLabel
        name="Add Note"
        aria-label="Add Note"
        onChange={this.onChange}
        value={this.state.newNote}
      />
      <div className="comment-save-button-container">
        <span className="cf-right-side">
          { this.props.job.user && this.sendToUserCheckbox() }
          <Button
            name="save"
            disabled={this.isSaveDisabled()}
            onClick={this.onSave}>
              Add Note
          </Button>
        </span>
      </div>
    </div>;
  }

}

NewNoteForm.propTypes = {
  job: PropTypes.object,
  notes: PropTypes.object,
  onSave: PropTypes.func,
  onCancel: PropTypes.func
};

export default class JobNotes extends React.PureComponent {

  render = () => {
    const { notes, job } = this.props;

    return <div className="job-notes">
      <h3>Notes</h3>
      <NewNoteForm job={job} notes={this} />
      <div>
        { notes.map((note, index) => {
          return <div className="job-note" key={`job-note-container-${index}`}>
            <div
              className="job-note-details"
              data-key={`job-note-${index}`}
              key={`job-note-${index}`}
              id={`job-note-${note.id}`}>

              <div>
                <span className="job-note-time">{moment(note.created_at).format(DATE_TIME_FORMAT)}</span>
                <span className="job-note-user">{note.user}</span>
              </div>
              <div className="job-note-note"><ReactMarkdown source={note.note} /></div>

            </div>
          </div>;
        })}
      </div>
    </div>;
  }
}

JobNotes.propTypes = {
  job: PropTypes.object,
  notes: PropTypes.array
};
