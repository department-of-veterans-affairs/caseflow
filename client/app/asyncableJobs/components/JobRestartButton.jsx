import React from 'react';
import Button from '../../components/Button';
import ApiUtil from '../../util/ApiUtil';
import PropTypes from 'prop-types';

class JobRestartButton extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      restarted: false,
      restarting: false
    };
  }

  restartJob = () => {
    this.setState({ restarting: true });
    this.sendRestart();
  }

  sendRestart = () => {
    const button = this;
    const page = this.props.page;
    const job = this.props.job;
    const url = `/asyncable_jobs/${job.klass}/jobs/${job.id}`;

    ApiUtil.patch(url, {}).
      then(
        (response) => {
          const responseObject = JSON.parse(response.text);

          Object.assign(job, responseObject);

          job.error = '';
          job.restarted = true;

          button.setState({
            restarted: true,
            restarting: false
          });

          // trigger table rerender
          const jobsRestarted = page.state.restarted + 1;

          page.setState({ restarted: jobsRestarted });
        }
      );
  }

  getButtonClassNames = () => {
    let classNames = ['usa-button'];

    if (this.state.restarting || this.state.restarted) {
      classNames.push('usa-button-disabled');
    }

    return classNames;
  }

  getButtonText = () => {
    let txt = 'Restart';

    if (this.state.restarting) {
      txt = 'Restarting';
    } else if (this.state.restarted) {
      txt = 'Restarted';
    } else if (this.props.job.processed_at) {
      txt = 'Processed';
    } else if (this.disableRestart()) {
      txt = 'Queued';
    }

    return txt;
  }

  disableRestart = () => {
    const job = this.props.job;

    if (job.processed_at) {
      return true;
    }

    if (!job.attempted_at) {
      return true;
    }

    const fiveMinutes = 300000;

    const lastAttempted = new Date(job.attempted_at).getTime();
    const submittedAt = new Date(job.last_submitted_at).getTime();
    const now = new Date().getTime();

    if ((now - lastAttempted) < fiveMinutes || (now - submittedAt) < fiveMinutes) {
      return true;
    }

    return false;
  }

  render = () => {
    const job = this.props.job;

    return <Button
      id={`job-${job.klass}-${job.id}`}
      title={`${job.klass} ${job.id}`}
      loading={this.state.restarting}
      loadingText="Restarting..."
      disabled={this.disableRestart()}
      onClick={() => {
        this.restartJob();
      }}
      classNames={this.getButtonClassNames()}
    >{this.getButtonText()}</Button>;
  }
}

JobRestartButton.propTypes = {
  job: PropTypes.object,
  page: PropTypes.object
};

export default JobRestartButton;
