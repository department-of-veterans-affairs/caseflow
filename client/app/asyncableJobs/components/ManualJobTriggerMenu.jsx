import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../../components/Button';
import StringUtil from '../../util/StringUtil';
import classNames from 'classnames';
import LoadingContainer from '../../components/LoadingContainer';
import { LOGO_COLORS } from '../../constants/AppConstants';

const ManualJobTriggerMenu = (props) => {

  const [loading, setLoading] = useState(false);
  const [manualJobType, setManualJobType] = useState(null);

  useEffect(() => {
    setManualJobType(props.manualJobType);
    if (props.manualJobType) {
      setLoading(false);
    }

  }, [props.manualJobType]);

  const renderMessage = () => {
    if (manualJobType) {
      const manualJobBannerClasses = classNames({
        success: props.manualJobSuccess,
        fail: !props.manualJobSuccess,
      }, 'manual-job-banner');

      const status = props.manualJobSuccess ? 'succeeded' : 'failed';

      return (
        <div className={manualJobBannerClasses}>
          Manual run of <strong>{StringUtil.snakeCaseToCapitalized(manualJobType)}</strong> has{' '}
          <strong>{status}</strong>
        </div>
      );
    }

    return null;
  };

  return (
    <div className="tab-border scheduled-jobs-tab">
      <br />
      <h1>Manually Perform Scheduled Async Jobs</h1>
      {loading &&
        <LoadingContainer color={LOGO_COLORS.QUEUE.ACCENT}>
          <div className="loading-div">
            Job is Running...
          </div>
        </LoadingContainer>
      }
      {!loading && (<>
        {renderMessage()}
        <table>
          <tbody className="manual-job-table">
            {props.supportedJobs.map((jobType) => (
              <tr key={`${jobType}-row`}>
                <td>
                  <h3>{StringUtil.snakeCaseToCapitalized(jobType)}</h3>
                </td>
                <td>
                  <Button
                    name={`trigger-${jobType}-job`}
                    onClick={() => {
                      setLoading(true);
                      props.sendJobRequest(jobType);
                    }}
                  >
                Perform Now
                  </Button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </>)}
    </div>
  );
};

ManualJobTriggerMenu.propTypes = {
  supportedJobs: PropTypes.arrayOf(
    PropTypes.string
  ),
  sendJobRequest: PropTypes.func.isRequired,
  manualJobStatus: PropTypes.number,
  manualJobSuccess: PropTypes.bool,
  manualJobType: PropTypes.string,
};

export default ManualJobTriggerMenu;
