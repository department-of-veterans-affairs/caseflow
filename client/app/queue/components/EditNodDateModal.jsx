import React, { useEffect, useState } from 'react';
import ReactMarkdown from 'react-markdown';
import { css } from 'glamor';
import PropTypes from 'prop-types';
import Modal from 'app/components/Modal';
import DateSelector from 'app/components/DateSelector';
import COPY from 'app/../COPY';
import { useDispatch, useSelector } from 'react-redux';
import { Controller, useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from 'yup';
import {
  resetSuccessMessages,
  showSuccessMessage,
} from '../uiReducer/uiActions';
import { editAppeal, editNodDateUpdates } from '../QueueActions';
import ApiUtil from '../../util/ApiUtil';
import { sprintf } from 'sprintf-js';
import { formatDateStr, DateString } from '../../util/DateUtil';
import { appealWithDetailSelector } from '../selectors';
import Alert from 'app/components/Alert';
import SearchableDropdown from 'app/components/SearchableDropdown';

const alertStyling = css({
  marginBottom: '2em',
  '& .usa-alert-text': { lineHeight: '1' },
});

export const changeReasons = [
  { label: 'New Form/Information Received', value: 'new_info' },
  { label: 'Data Entry Error', value: 'entry_error' },
];

export const EditNodDateModalContainer = ({ onCancel,
  onSubmit,
  nodDate: origNodDate,
  appealId,
  reason: origReason }) => {
  const [showTimelinessError, setTimelinessError] = useState(false);
  const [issues, setIssues] = useState(null);

  const dispatch = useDispatch();
  const appeal = useSelector((state) =>
    appealWithDetailSelector(state, { appealId })
  );

  useEffect(() => {
    dispatch(resetSuccessMessages());
  }, []);

  const handleCancel = () => onCancel();

  const handleSubmit = ({ nodDate, reason }) => {
    const alertInfo = {
      appellantName: appeal.appellantFullName,
      nodDateStr: formatDateStr(origNodDate, 'YYYY-MM-DD', 'MM/DD/YYYY'),
      receiptDateStr: formatDateStr(nodDate, 'YYYY-MM-DD', 'MM/DD/YYYY'),
    };

    const title = COPY.EDIT_NOD_DATE_SUCCESS_ALERT_TITLE;
    const detail = sprintf(COPY.EDIT_NOD_DATE_SUCCESS_ALERT_MESSAGE, alertInfo);

    const successMessage = {
      title,
      detail,
    };
    const payload = {
      data: {
        receipt_date: nodDate,
        change_reason: reason,
      },
    };

    ApiUtil.patch(`/appeals/${appealId}/nod_date_update`, payload).then((data) => {
      dispatch(editAppeal(appealId, {
        nodDate: data.body.nodDate,
        docketNumber: data.body.docketNumber,
        reason: data.body.changeReason
      }));

      if (data.body.affectedIssues) {
        setIssues({ affectedIssues: data.body.affectedIssues, unaffectedIssues: data.body.unaffectedIssues });
        setTimelinessError(true);
      } else {
        dispatch(editNodDateUpdates(appealId, data.body.nodDateUpdate));
        dispatch(showSuccessMessage(successMessage));
        onSubmit?.();
        window.scrollTo(0, 0);
      }
    });
  };

  return (
    <EditNodDateModal
      onCancel={handleCancel}
      onSubmit={handleSubmit}
      nodDate={origNodDate}
      reason={origReason}
      appealId={appealId}
      appellantName={appeal.appellantFullName}
      showTimelinessError={showTimelinessError}
      issues={issues}
    />
  );
};

export const EditNodDateModal = ({
  onCancel,
  onSubmit,
  nodDate,
  reason,
  showTimelinessError,
  issues
}) => {
  const [showWarning, setWarning] = useState(false);

  yup.addMethod(yup.date, 'isLater', function () {
    return this.test((value) => {
      const formattedValue = formatDateStr(value, 'YYYY-MM-DD', 'YYYY-MM-DD');
      const formattedNodDate = formatDateStr(nodDate, 'YYYY-MM-DD', 'YYYY-MM-DD');

      if (formattedValue > formattedNodDate) {
        setWarning(true);
      } else {
        setWarning(false);
      }

      return value;
    });
  });

  const schema = yup.object().shape({
    nodDate: yup.
      date().
      min('2018-01-01', COPY.EDIT_NOD_DATE_PRE_AMA_DATE_ERROR_MESSAGE).
      max(new Date(), COPY.EDIT_NOD_DATE_FUTURE_DATE_ERROR_MESSAGE).
      isLater().
      typeError('Invalid date.').
      required(),
    reason: yup.
      string().
      typeError('Required.').
      required().
      oneOf(changeReasons.map((changeReason) => changeReason.value))
  });

  const { register, errors, control, handleSubmit } = useForm(
    {
      defaultValues: { nodDate, reason },
      resolver: yupResolver(schema),
      mode: 'onChange',
      reValidateMode: 'onChange'
    }
  );

  const buttons = [];

  if (!showTimelinessError) {
    buttons.push({
      classNames: ['cf-modal-link', 'cf-btn-link'],
      name: 'Cancel',
      onClick: onCancel
    });
  }

  buttons.push({
    classNames: ['usa-button', 'usa-button-primary'],
    name: showTimelinessError ? 'Close' : 'Submit',
    onClick: showTimelinessError ? onCancel : handleSubmit(onSubmit)
  });

  let modalContent;

  if (showTimelinessError) {
    modalContent = <div>
      { showTimelinessError ? <Alert
        message={COPY.EDIT_NOD_DATE_TIMELINESS_ERROR_MESSAGE}
        styling={alertStyling}
        title={COPY.EDIT_NOD_DATE_TIMELINESS_ALERT_TITLE}
        type="error"
        scrollOnAlert={false}
      /> : null }

      <strong>Affected Issue(s)</strong>
      <ul className="cf-error">
        {issues.affectedIssues.map((issue) => {

          return <li key={issue.id}>
            {issue.description}
            <div>
              (Decision Date: <DateString date={issue.approx_decision_date} dateFormat = "MM/DD/YYYY" />)
            </div>
          </li>;
        })}
      </ul>

      { issues.unaffectedIssues.length > 0 && <strong>Unaffected Issue(s)</strong> }
      <ul>
        {issues.unaffectedIssues.map((issue) => {

          return <li key={issue.id}>
            {issue.description}
            <div>
              (Decision Date: <DateString date={issue.approx_decision_date} dateFormat = "MM/DD/YYYY" />)
            </div>
          </li>;
        })}
      </ul>
      <br />
      <div>
        <ReactMarkdown source={COPY.EDIT_NOD_DATE_TIMELINESS_COB_MESSAGE} />
      </div>
    </div>;
  } else {
    modalContent = <div>
      <div>
        <ReactMarkdown source={COPY.EDIT_NOD_DATE_MODAL_DESCRIPTION} />
      </div>
      { showWarning ? <Alert
        message={COPY.EDIT_NOD_DATE_WARNING_ALERT_MESSAGE}
        styling={alertStyling}
        title=""
        type="info"
        scrollOnAlert={false}
      /> : null }
      <form onSubmit={handleSubmit(onSubmit)}>
        <DateSelector
          inputRef={register}
          name="nodDate"
          errorMessage={errors.nodDate?.message}
          label={COPY.EDIT_NOD_DATE_LABEL}
          strongLabel
          type="date"
        />
        <Controller
          name="reason"
          control={control}
          defaultValue={null}
          render={({ onChange, ...rest }) => (
            <SearchableDropdown
              {...rest}
              label="Reason for edit"
              errorMessage={errors.reason?.message}
              placeholder="Select the reason..."
              options={changeReasons}
              debounce={250}
              strongLabel
              onChange={(valObj) => onChange(valObj?.value)}
            />
          )}
        />
      </form>
    </div>;
  }

  return (
    <Modal
      title={COPY.EDIT_NOD_DATE_MODAL_TITLE}
      onCancel={onCancel}
      onSubmit={onSubmit}
      closeHandler={onCancel}
      buttons={buttons}>
      {modalContent}
    </Modal>
  );
};

EditNodDateModalContainer.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  nodDate: PropTypes.string.isRequired,
  reason: PropTypes.object,
  appealId: PropTypes.string.isRequired,
};

EditNodDateModal.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  nodDate: PropTypes.string.isRequired,
  reason: PropTypes.object,
  appealId: PropTypes.string.isRequired,
  showTimelinessError: PropTypes.bool.isRequired,
  issues: PropTypes.object
};
