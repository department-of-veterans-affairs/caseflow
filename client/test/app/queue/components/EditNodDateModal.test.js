import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { EditNodDateModal } from 'app/queue/components/EditNodDateModal';

Enzyme.configure({ adapter: new Adapter() });

describe('EditNodDateModal', () => {
  const onSubmit = jest.fn();
  const onCancel = jest.fn();
  const defaultNodDate = '2020-10-31';
  const appealId = 'tb78ti7in77n';
  const showTimelinessError = false;
  const defaults = {
    appealId,
    onSubmit,
    onCancel,
    nodDate: defaultNodDate,
    showTimelinessError
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { container } = render(<EditNodDateModal {...defaults} />);

    expect(container).toMatchSnapshot();
  });

  it('should fire cancel event', async () => {
    render(<EditNodDateModal {...defaults} />);

    expect(onCancel).not.toHaveBeenCalled();

    await userEvent.click(screen.getByRole('button', { name: /cancel/i }));

    expect(onCancel).toHaveBeenCalled();
  });
});
