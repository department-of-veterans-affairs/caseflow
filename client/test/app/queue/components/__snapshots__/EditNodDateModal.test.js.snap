// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNodDateModal renders correctly 1`] = `
<div>
  <section
    aria-describedby="modal_id-desc"
    aria-labelledby="modal_id-title"
    aria-modal="true"
    class="cf-modal active "
    id="modal_id"
    role="dialog"
  >
    <div
      class="cf-modal-body"
      id=""
    >
      <button
        class="cf-modal-close"
        id="Edit-NOD-Date-button-id-close"
        type="button"
      >
        <span
          class="usa-sr-only"
        >
          Close
        </span>
        <svg
          class="cf-icon-close"
          height="55"
          viewBox="0 0 55 55"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            close
          </title>
          <g
            fill="#212121"
          >
            <path
              d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
            />
          </g>
        </svg>
      </button>
      <div
        style="display: flex;"
      >
        <div
          data-css-1gs0ko2=""
        >
          <h1
            id="modal_id-title"
          >
            Edit NOD Date
          </h1>
          <div
            data-css-o98ubo=""
          >
            <div>
              <div>
                <p>
                  <strong>
                    Editing the NOD date affects the time it takes for a Veteran to receive a response to their appeal.
                  </strong>
                </p>
              </div>
              <form>
                <div
                  class="cf-form-textinput   "
                >
                  <label
                    for="nodDate"
                  >
                    <strong>
                      <span>
                        NOD Date
                      </span>
                    </strong>
                  </label>
                  <input
                    class="cf-form-textinput"
                    id="nodDate"
                    max="9999-12-31"
                    name="nodDate"
                    placeholder="mm/dd/yyyy"
                    type="date"
                    value=""
                  />
                </div>
                <div
                  class=""
                >
                  <div
                    class="cf-form-dropdown dropdown-reason"
                  >
                    <label
                      class="question-label"
                      for="reason"
                      id="reason-label"
                    >
                      <strong>
                        <span>
                          Reason for edit
                        </span>
                      </strong>
                    </label>
                    <div
                      class="cf-select"
                    >
                      <div
                        class=" css-2b097c-container"
                      >
                        <div
                          class="cf-select__control css-yk16xz-control"
                        >
                          <div
                            class="cf-select__value-container css-g1d714-ValueContainer"
                          >
                            <div
                              class="cf-select__placeholder css-1wa3eu0-placeholder"
                            >
                              Select the reason...
                            </div>
                            <div
                              class="css-81nyic-Input"
                            >
                              <div
                                class="cf-select__input"
                                style="display: inline-block;"
                              >
                                <input
                                  aria-autocomplete="list"
                                  aria-expanded="false"
                                  aria-haspopup="true"
                                  aria-labelledby="reason-label"
                                  aria-owns="reason-listbox"
                                  autocapitalize="none"
                                  autocomplete="off"
                                  autocorrect="off"
                                  id="reason"
                                  role="combobox"
                                  spellcheck="false"
                                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                                  tabindex="0"
                                  type="text"
                                  value=""
                                />
                                <div
                                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-6q0nyr-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="reason"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="cf-modal-divider"
      />
      <div
        class="cf-modal-controls"
      >
        <span>
          <button
            class="cf-modal-link cf-btn-link cf-push-left usa-button"
            id="Edit-NOD-Date-button-id-0"
            type="button"
          >
            Cancel
          </button>
        </span>
        <span>
          <button
            class="usa-button usa-button-primary cf-push-right usa-button"
            id="Edit-NOD-Date-button-id-1"
            type="button"
          >
            Submit
          </button>
        </span>
      </div>
    </div>
  </section>
</div>
`;
